src/compiler/scala/tools/nsc/typechecker/Implicits.scala

```diff
@@ -447,12 +451,27 @@ trait Implicits {
         reporter.echo(pos, s"$what is not a valid implicit value for $pt because:\n$reason")
       SearchFailure
     }
+
+    /** Check a case when `info1` is `ArrayCharSequence` and `info2` is `charArrayOps` from `scala.Predef`
+     *
+     * It can be true only when both this methods are imlicit the same type
+     * and it is a case when both are implemented `isEmpty`
+     * because `java.lang.CharSequence` has `isEmpty` since JDK 15
+     *
+     * Force to select scala implementation of `isEmpty` from `charArrayOps`
+     * because it produces bytecode that can be run on JDK before 15
+     */
+    def resolveJDK15CharSequenceConflict(info1: ImplicitInfo, info2: ImplicitInfo) =
+      info1.pre =:= info2.pre &&
+        info1.name == termArrayCharSequence && info2.name == termCharArrayOps &&
+        info1.pre.termSymbol.fullName == "scala.Predef"
+
     /** Is implicit info `info1` better than implicit info `info2`?
      */
     def improves(info1: ImplicitInfo, info2: ImplicitInfo) = {
       if (StatisticsStatics.areSomeColdStatsEnabled) statistics.incCounter(improvesCount)
       (info2 == NoImplicitInfo) ||
-      (info1 != NoImplicitInfo) && {
+      (info1 != NoImplicitInfo) && resolveJDK15CharSequenceConflict(info1, info2) || {
```

Comment:
would be good for performance to do this only after checking the `improvesCache`